modules/internal/ChapelSyncvar.chpl

```diff
@@ -168,6 +168,11 @@ module ChapelSyncvar {
     return b;
   }
 
+  // Do not allow implicit reads of sync/single vars.
+  proc _syncvar.readThis(x: Reader) {
+    compilerError("sync/single variables cannot currently be read - use writeEF/writeFF instead");
```

Comment:
Would it be better to indicate just the relevant case, i.e. sync variables cannot currently be read versus sync/single?